optimum/neuron/modeling_seq2seq.py

```diff
@@ -0,0 +1,912 @@
+        stopping_criteria: Optional[StoppingCriteriaList] = None,
+        prefix_allowed_tokens_fn: Optional[Callable[[int, torch.Tensor], List[int]]] = None,
+        assistant_model: Optional["PreTrainedModel"] = None,
+        streamer: Optional["BaseStreamer"] = None,
+        num_return_sequences: Optional[int] = None,
+        **kwargs,
+    ):
+        max_length = self.neuron_configs[ENCODER_NAME].sequence_length
+        num_beams = self.neuron_configs[ENCODER_NAME].num_beams
+        batch_size = self.neuron_configs[ENCODER_NAME].batch_size
+
+        inputs = {"input_ids": input_ids}
+        if attention_mask is not None:
+            inputs["attention_mask"] = attention_mask
+        inputs = self._pad_to_compiled_shape(inputs)
+
+        past_key_values = self.encoder(**inputs)
+
+        decoder_attention_mask = torch.cat(
+            [torch.zeros((batch_size, max_length - 1), dtype=torch.int64), torch.ones((1, 1), dtype=torch.int64)],
+            axis=1,
+        )
+
+        # copy the new cache state to the decoder
+        for state, tensor in zip(self.decoder.model.parameters(), past_key_values):
+            state.copy_(tensor)
+
+        output = super().generate(
+            **inputs,
+            generation_config=generation_config,
+            logits_processor=logits_processor,
+            stopping_criteria=stopping_criteria,
+            prefix_allowed_tokens_fn=prefix_allowed_tokens_fn,
+            assistant_model=assistant_model,
+            streamer=streamer,
+            num_return_sequences=num_return_sequences,
+            max_length=max_length,
+            num_beams=num_beams,
+            do_sample=kwargs.pop("do_sample", False),
+            use_cache=kwargs.pop("use_cache", True),
+            decoder_attention_mask=decoder_attention_mask,
+            # Pass fake encoder_outputs so the transfomers code will not invoke the encoder
+            encoder_outputs={"last_hidden_state": torch.ones((batch_size, max_length, 1))},
+        )
+        return output
+
+    def beam_search(
+        self,
+        input_ids: torch.LongTensor,
+        beam_scorer: "BeamScorer",
+        logits_processor: Optional[LogitsProcessorList] = None,
+        stopping_criteria: Optional[StoppingCriteriaList] = None,
+        pad_token_id: Optional[int] = None,
+        eos_token_id: Optional[Union[int, List[int]]] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        output_scores: Optional[bool] = None,
+        return_dict_in_generate: Optional[bool] = None,
+        synced_gpus: Optional[bool] = False,
+        seq_length: Optional[int] = None,
+        **model_kwargs,
+    ) -> Union[BeamSearchOutput, torch.LongTensor]:
+        """
+        Overriding beam search to use next_token_scores returned from neuron device instead of logits.
+        """
+        logits_processor = logits_processor if logits_processor is not None else LogitsProcessorList()
```

Comment:
You never apply the processors here. This means that a lot of generation options will actually be ignored (like repetition_penalty for instance). We should warn the user that this is not supported.

Reply:
Yeah, the `next_token_scores` and `topk` are already calculated inside the compiled decoder, so the `logits_processor` is not used here... It's something that we can let user decide during the compilation instead of during the inference...And I might not add the compilation support taking it into consideration in this PR neither, maybe in a later PR. 

I add a warning to explain this and ask people to file an issue if they want us to support it.